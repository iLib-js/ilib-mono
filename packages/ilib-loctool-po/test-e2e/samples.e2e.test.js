const path = require("path");
const { expectFileToMatchSnapshot, LoctoolRunner } = require("@ilib-mono/e2e-test");

describe("samples", () => {
    describe("po", () => {
        const projectPath = path.resolve(__dirname, "..", "samples", "po");

        beforeAll(async () => {
            const loctool = new LoctoolRunner(projectPath);
            await loctool.run("localize");
        });

        it("should produce an extracted XLIFF file", () => {
            const xliffPath = path.resolve(projectPath, "xliffs", "sample-po-extracted.xliff");
            expectFileToMatchSnapshot(xliffPath);
        });
    });
});
