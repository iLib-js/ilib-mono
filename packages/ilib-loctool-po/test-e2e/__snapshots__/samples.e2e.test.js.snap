// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`samples po should produce an extracted XLIFF file 1`] = `
"<?xml version="1.0" encoding="utf-8"?>
<xliff version="1.2">
  <file original="" source-language="en-US" product-name="sample-po">
    <body>
      <trans-unit id="1" resname="one string" restype="plural" datatype="po" extype="one">
        <source>one string</source>
        <note>{"translator":["a plural string"],"pluralForm":"one","pluralFormOther":"one string"}</note>
      </trans-unit>
      <trans-unit id="2" resname="one string" restype="plural" datatype="po" extype="other">
        <source>{$count} strings</source>
        <note>{"translator":["a plural string"],"pluralForm":"other","pluralFormOther":"one string"}</note>
      </trans-unit>
      <trans-unit id="3" resname="Good bye" restype="string" datatype="po">
        <source>Good bye</source>
        <note>{"translator":["another string"]}</note>
      </trans-unit>
      <trans-unit id="4" resname="Thank you too" restype="string" datatype="po">
        <source>Thank you too</source>
        <note>{"translator":["string with continuation"]}</note>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;
